src/workerd/api/basics.c++

```diff
@@ -276,8 +278,16 @@ void EventTarget::addEventListener(jsg::Lock& js,
                 removeEventListener(js, kj::mv(type), kj::mv(handler), kj::none);
               });
 
-      // TODO(now): Attaching a strong reference to signal probably isn't correct; it fixes the
-      // memory error in AbortSignal::any(), but probably by creating a leaked cycle?
+      // The returned native handler captures a bare reference to signal and
+      // will be held by this EventTarget. The signal is the only thing that
+      // triggers it. If signal is gc'd the native handler created here could
+      // still be alive which means *technically* it will be holding a bare
+      // reference for something that is already destroyed. However, there's
+      // nothing else that would trigger it so it's generally safe-ish. That
+      // said, it's still a potential UAF so let's guard against it by attaching
+      // a strong reference to the signal to the event handler. This will mean
+      // likely keeping the signal in memory longer if it can otherwise be
+      // gc'd but that's ok, the impact should be minimal.
       return signal->newNativeHandler(js, kj::str("abort"), kj::mv(func), true)
           .attach(signal.addRef());
```

Comment:
Huh... I'm surprised this was actually a correct thing to do...

I was assuming that doing `.attach()` here was the wrong thing, in part because it adds a reference between two `jsg::Object`s (a reference from `EventTarget` to `signal`) that the V8 garbage collector can't see, so could potentially create cycles that it couldn't easily collect, and that we'd need to actually use something more like `JSG_VISITABLE_LAMBDA()`.

(However, I did try adding it directly to the `JSG_VISITABLE_LAMBDA()` above, but that didn't seem to ensure that `signal` remained alive.)

Reply:
That said... I think potentially cycle-prone code is an improvement over potentially memory-unsafe code, so probably worth committing as-is and addressing in a followup pass.

Reply:
Hmm you might be right. I'll look at that in the morning